pkgs/by-name/br/breakpointHook/package.nix

```diff
@@ -1,6 +1,37 @@
-{ stdenv, makeSetupHook }:
+{
+  lib,
+  stdenv,
+
+  bash,
+  bashInteractive,
+  coreutils,
+  makeSetupHook,
+  procps,
+  util-linux,
+  writeShellScriptBin,
+}:
+
+let
+  attach = writeShellScriptBin "attach" ''
```

Comment:
I'd say pkgs.writeShellApplication is preferred here. It will also run `shellcheck` on top and already has an argument for extending $PATH.

Reply:
I specifically chose not to use writeShellApplication.I did implement it using it before, but then ended up setting the PATH manually as writeShellApplication just prepends it, which doesn't prevent you from forgetting dependencies, which is the whole point IMO.
Also, I disabled the shellcheck, because it significantly increases eval time as it pulls in the whole haskell toolchain.
Depending on haskell would also make it annoying to use breakpointHook on staging.
I'm happy to use `writeShellApplication` but I need to fix it first, so it supports a `pure` mode whre the PATH is fully overridden. Should I do this in this PR?